### ISSUE TITLE
near_text is very slow to calculate embedding with text2vec-transformers module

### ISSUE THREAD

**Issue by bfichter** [weaviate/weaviate#4062]
### How to reproduce this bug?

I'm running `semitechnologies/weaviate:1.23.2` locally with a singular t2v-transformers module as a vectorizer, using the pre-built `semitechnologies/transformers-inference:distilbert-base-uncased`. This is running on a macbook pro with an M2 chip.

When inserting a new object via `collection.data.insert(properties)`, the call (and underlying embedding calculation) is very fast. However, when calling `collection.query.near_text(query)`, there is a significant amount of latency (~4 seconds) before the vector is calculated and returned from the vectorizer module, and once it's returned the actual query is very fast to run. 

This behavior is observable even if the collection is totally empty (i.e. it doesn't just arise after the collection gets to a certain size).

### What is the expected behavior?

The time to calculate an embedding should be relatively constant, regardless of the high level function being called (i.e. `insert` vs `near_text`. And embeddings should not take multiple seconds to be computed.

### What is the actual behavior?

Embeddings are calculated quickly in `insert`, but extremely slowly in `near_text`.

### Supporting information

This is my docker-compose.yml:

```
---
version: '3.4'
services:
  weaviate:
    command:
    - --host
    - 0.0.0.0
    - --port
    - '8080'
    - --scheme
    - http
    image: semitechnologies/weaviate:1.23.2
    ports:
    - 8080:8080
    - 50051:50051
    volumes:
    - weaviate_data:/var/lib/weaviate
    restart: on-failure:0
    environment:
      TRANSFORMERS_INFERENCE_API: 'http://t2v-transformers:8080'
      QUERY_DEFAULTS_LIMIT: 25
      AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED: 'true'
      PERSISTENCE_DATA_PATH: '/var/lib/weaviate'
      DEFAULT_VECTORIZER_MODULE: 'text2vec-transformers'
      ENABLE_MODULES: 'text2vec-transformers'
      CLUSTER_HOSTNAME: 'node1'
  t2v-transformers:
    image: 'semitechnologies/transformers-inference:distilbert-base-uncased'
volumes:
  weaviate_data:
...

### Server Version

1.23.2 (running locally with docker compose)

### Code of Conduct

- [X] I have read and agree to the Weaviate's [Contributor Guide](https://weaviate.io/developers/contributor-guide) and [Code of Conduct](https://weaviate.io/service/code-of-conduct)

**Comment by bfichter** [comment]
Pretty sure this is an issue with [t2v-transformers-models](https://github.com/weaviate/t2v-transformers-models/tree/270d004e7c9537b63f669189decce78e5bed9686). 

It seems if that module gets multiple concurrent requests, it becomes _really_ slow to respond. If I just run that container independently, and make multiple requests to POST /vectors, it quickly chokes and takes multiple seconds or longer to respond to any request.

Might be an issue with this [scheduling logic](https://github.com/weaviate/t2v-transformers-models/blob/270d004e7c9537b63f669189decce78e5bed9686/vectorizer.py#L51)?

**Comment by etiennedi** [comment]
Good catch, I think we should move the issue over there. AFAIK @antas-marcin is aware of the single-threaded nature and parallelism restrictions of the transformers inference containers and has plans to resolve them in the future.

**Comment by netapy** [comment]
@bfichter I made Text-Embedding-Inference by HuggingFace compatible with Weaviate as a drop in replacement for t2v-transformers . Works very well on my end.

https://github.com/OrdalieTech/text-embeddings-inference-4-weaviate

**Comment by dudanogueira** [comment]
Closing. If you still face this error, let us know.
